task6/index.js

```diff
@@ -0,0 +1,364 @@
+            createdAt: new Date('2021-03-20T11:00:00'),
+            link: 'https://sol.by/',
+            vendor: 'SOL',
+            photoLink: 'https://enguide.by/s/public/upload/images/e6ce/8bd6/f526/3246/470a/4b7d/f2ed/7df0.jpeg',
+            validUntil: new Date('2021-08-21T24:00:00'),
+            discount: '20%',
+            hashTags: ['Образовательные услуги']
+        }
+    ]
+
+    function getAds(skip = 0, top = 10, filterConfig = undefined) {
+        if (typeof skip != 'number' || typeof top != 'number') {
+            return 'Error'
+        }
+        let elem = adList;
+        if (filterConfig) {
+            for (let item in filterConfig) {
+                if (item === 'createdAt') {
+                    elem = elem.filter(item => item.createdAt <= filterConfig.createdAt);
+                }
+                if (item === 'validUntil') {
+                    elem = elem.filter(item => item.validUntil >= filterConfig.validUntil);
+                }
+                if (item === 'vendor') {
+                    elem = elem.filter(item => item.vendor === filterConfig.vendor);
+                }
+                else {
+                    return 'Error'
+                }
+            }
+        }
+        elem.sort(function (a, b) {
+            return a.createdAt - b.createdAt;
+        });
+        return elem.slice(skip, skip + top);
+    }
+
+    console.log(getAds(0, 10))
+    console.log(getAds(10, 10))
+    console.log(getAds('3'))
+    console.log(getAds(0, 4, {vendor: 'Анна Петрачкова'}))
+    console.log(getAds(0, 7, {validUntil: new Date('2021-05-01T24:00:00')}))
+
+
+    function getAd(id) {
+        return adList.filter(item => item.id == id)
+    }
+    console.log(getAd(12))
+
+    function addAd(newElem) {
+        if (validateAd(newElem)) {
+            for (let item of adList) {
+                if (item.id == newElem.id) {
+                    return 'Error'
+                }
+            }
+            adList.push(newElem)
+            return adList
+        }
+        else return false
+    }
+    console.log(addAd({id: '21',
+        description: 'Суши от Rollbox',
+        createdAt: new Date('2021-03-01T11:00:00'),
+        link: 'https://rollbox.by/',
+        vendor: 'Rollbox',
+        photoLink: 'https://sushichefarts.by/upload/iblock/ba7/ba79b531e47dfa1bec62fce7d1817916.jpg',
+        validUntil: new Date('2021-07-01T24:00:00'),
+        discount: '40%',
+        hashTags: ['Прочие услуги']}))
+
+    console.log(addAd({id: '14',
+        description: 'Суши от Rollbox',
+        createdAt: new Date('2021-03-01T11:00:00'),
+        link: 'https://rollbox.by/',
+        vendor: 'Rollbox',
+        photoLink: 'https://sushichefarts.by/upload/iblock/ba7/ba79b531e47dfa1bec62fce7d1817916.jpg',
+        validUntil: new Date('2021-07-01T24:00:00'),
+        discount: '40%',
+        hashTags: ['Прочие услуги']}))
+
+    function editAd(id, propertyChange, changedValue) {
```

Comment:
What if I would like to edit both description and link at the same time, for example? I'd rather advice to accept an object as a parameter here.